namespace-provisioner/how-tos.hbs.md

```diff
@@ -0,0 +1,254 @@
+
+Users can choose to maintain the [`desired-namespaces`](about.hbs.md#nsp-component-desired-namespaces-configmap) ConfigMap in their git repository and opt not to use the namespace-provisioner controller. Users can use their choice of GitOps tools to override the [`desired-namespaces`](about.hbs.md#nsp-component-desired-namespaces-configmap) ConfigMap in the tap-namespace-provisioning namespace. 
+
+>**NOTE:** you need to set the “controller” tap value key to “false” (Default is “true”) for namespace provisioner in tap-values.yaml file. The [`controller`](install.hbs.md#customized-installation) value in tap-values decides whether the Out-of-the-box controller is installed and used for maintaining the list of desired-namespaces or not.</br>
+If the controller is installed as part of the package, when it reconciles, it will remove all namespaces that are not properly labeled with the [`namespace_selector`](install.hbs.md#customized-install) specified in the tap values (Default label selector is **`apps.tanzu.vmware.com/tap-ns=""`**). 
+
+Use the following snippet as reference for the desired-namespaces ConfigMap that you can put on your git repository.
+Desired-namespaces.yaml ([Link to sample repo file](https://github.com/vmware-tanzu/application-accelerator-samples/blob/main/namespace-provisioner-gitops-examples/desired-namespaces/gitops-managed-desired-namespaces.yaml))
+
+```yaml
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: desired-namespaces
+  namespace: tap-namespace-provisioning
+  annotations:
+    kapp.k14s.io/create-strategy: fallback-on-update
+    namespace-provisioner.apps.tanzu.vmware.com/no-overwrite: "" #! This annotation tells the provisioner app to not override this configMap as this is your desired state.
+data:
+  namespaces.yaml: |
+    #@data/values
+    ---
+    namespaces:
+    - name: python-backend-app
+      language: python
+      scanpolicy: snyk
+    - name: angular-fe-app
+      language: angular
+
+```
+
+We will use kubectl to override this ConfigMap manually, but in a customer environment, this ConfigMap can be overridden with the customer’s tool of choice for GitOps. The method for overriding the [`desired-namespaces`](about.hbs.md#nsp-component-desired-namespaces-configmap) ConfigMap is out of scope for this package. 
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/vmware-tanzu/application-accelerator-samples/main/namespace-provisioner-gitops-examples/desired-namespaces/gitops-managed-desired-namespaces.yaml
+```
+
+When this change is applied, the provisioner app will start the reconcile process and provision the resources on the given namespaces.
+
+>**WARNING:** If there is a namespace in your GitOps repo `desired-namespaces` list that does not exist on the cluster, the `provisioner` app will fail to reconcile and will not be able to create resources. Additionally, the creation of the namespaces themselves is out of the scope for the namespace provisioner package.
```

Comment:
should there be a link to Legacy manual developer namespace setup instructions here

Reply:
I don't think its necessary to have the legacy manual developer ns setup link here.